tests.js

```diff
@@ -60,39 +60,79 @@ QUnit.module("Logo Unit Tests", {
 
     this.assert_equals = function(expression, expected) {
       var actual = this.interpreter.run(expression, {returnResult: true});
-      if (typeof expected === 'object') {
-        t.deepEqual(actual, expected, expression);
-      } else if (typeof expected === 'number' && typeof actual === 'number') {
-        t.ok(Math.abs(actual - expected) < EPSILON, expression);
-      } else {
-        t.strictEqual(actual, expected, expression);
-      }
+      var done = t.async();
+      actual.then(function (result) {
+        if (typeof expected === 'object') {
+          t.deepEqual(result, expected, expression);
+        } else if (typeof expected === 'number' && typeof result === 'number' &&
+                   (Math.floor(expected) != expected || Math.floor(result) != result)) {
+          t.ok(Math.abs(result - expected) < EPSILON, expression);
+        } else {
+          t.strictEqual(result, expected, expression);
+        }
+        done();
+      }, function (failure) {
+        t.strictEqual(failure, expected, expression);
+        done();
```

Comment:
Mentioned elsewhere, but you can move the redundant `done()` calls into a trailing `.then()` which behaves as a `finally` for promise chains.